include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -4665,7 +4671,128 @@ struct OrtApi {
-};
+
+  /** \brief Create OrtDevice object.
+   *
+   * \param[in] device_type
+   * \param[in] memory_type
+   * \param[in] device_id
+   * \param[out] out OrtDevice object
+   *
+   * \since Version 1.xx.
+   */
+  ORT_API2_STATUS(CreateDevice, _In_ enum OrtMemoryInfoDeviceType device_type, _In_ enum OrtMemoryType memory_type, _In_ int16_t device_id, _Outptr_ const OrtDevice** out);
+
+  /** \brief Get OrtMemoryInfoDeviceType property from OrtDevice object.
+   *
+   * \param[in] device OrtDevice object
+   * \param[out] out OrtMemoryInfoDeviceType property
+   *
+   * \since Version 1.xx.
+   */
+  ORT_API2_STATUS(DeviceGetType, _In_ const OrtDevice* device, _Out_ OrtMemoryInfoDeviceType* out);
+
+  /** \brief Get OrtMemoryType property from OrtDevice object.
+   *
+   * \param[in] device OrtDevice object
+   * \param[out] out OrtMemoryType property
+   *
+   * \since Version 1.xx.
+   */
+  ORT_API2_STATUS(DeviceGetMemoryType, _In_ const OrtDevice* device, _Out_ OrtMemoryType* out);
+
+  /** \brief Get device id property from OrtDevice object.
+   *
+   * \param[in] device OrtDevice object
+   * \param[out] out device id property
+   *
+   * \since Version 1.xx.
+   */
+  ORT_API2_STATUS(DeviceGetId, _In_ const OrtDevice* device, _Out_ int16_t* out);
+
+  /** \brief Release OrtDevice object.
+   *
+   * \since Version 1.xx.
+   */
+  ORT_CLASS_RELEASE(Device);
+
+  /** \brief Register the plugin ExecutionProvider library
+   *
+   * The plugin ExecutionProvider library will be loaded and EP factory object will be created and saved in OrtEnv object
+   *
+   * \param[in] lib_path the path of the plugin ExecutionProvider library
+   * \param[in] env OrtEnv object
+   * \param[in] ep_name the plugin ExecutionProvider name
+   *
+   * \since Version 1.xx.
+   */
+  ORT_API2_STATUS(RegisterPluginExecutionProviderLibrary, _In_ const ORTCHAR_T* lib_path, _In_ OrtEnv* env, _In_ const char* ep_name);
+
+  /** \brief Unregister the plugin ExecutionProvider library
+   *
+   * The plugin ExecutionProvider factory will be removed from OrtEnv object
+   *
+   * \param[in] env OrtEnv object
+   * \param[in] ep_name the plugin ExecutionProvider name
+   *
+   * \since Version 1.xx.
+   */
+  ORT_API2_STATUS(UnregisterPluginExecutionProviderLibrary, _In_ OrtEnv* env, _In_ const char* ep_name);
+
+  /** \brief Append the plugin ExecutionProvider factory into the session option with provider options
+   *
+   * \param[in] options OrtSessionOptions object
+   * \param[in] ep_name the plugin ExecutionProvider name
+   * \param[in] env OrtEnv object
+   * \param[in] provider_options_keys provider options' keys
+   * \param[in] provider_options_values provider options' values
+   * \param[in] num_keys the number of the provider options' key-value pairs
+   *
+   * \since Version 1.xx.
+   */
+  ORT_API2_STATUS(SessionOptionsAppendPluginExecutionProvider, _In_ OrtSessionOptions* options, _In_ const char* ep_name, _In_ OrtEnv* env,
```

Comment:
Can this can be removed? We may be able to use the existing C API function called SessionOptionsAppendExecutionProvider. The existing C API does not take an OrtEnv parameter, but we can just get the default OrtEnv since there is only one per process.